ldms/src/store/avro_kafka/store_avro_kafka.c

```diff
@@ -60,7 +61,7 @@ typedef struct aks_handle_s
 static const char *_help_str =
     "   config name=store_avro_kafka [path=JSON_FILE]\n"
     "       encoding=MODE"
-    "            MODE is one of JSON or AVRO (default)."
+    "            MODE is one of JSON or YYJSON or AVRO (default)."
```

Comment:
I don't think this should be a configurable mode. yyjson is a json implementation, like libjansson, not a different encoding. It somes out as json regardless of the library used to to do the encoding.

If yyjson is going to be a _required_ external library (and that _is_ how this patch implements it in configure.ac), and it is a lot faster, then I would say just replace libjansson with yyjson.

On the other hand, if we want to make yyjson an optional external library, then we should have conditional compilation of the backend that implements the json mode. If only libjansson is available, compile against that. If yyjson is available, compile against that.

Reply:
That's reasonable.  The existing plugin doesn't use jansson (which in our tests is actually slower than the sprintf-based code in the ldmsd_decomp.c), at least using the json encoding mode.  

If you can give me some guidance about libtool library presence testing and what defines it puts in place, I can change it so that it's conditional to libyyjson being on the system.  It should be pretty easy to make it a seamless replacement.